src/client/display_configuration.cpp

```diff
@@ -55,7 +55,11 @@ mcl::DisplayOutput::DisplayOutput(size_t num_modes_, size_t num_formats)
 
 mcl::DisplayOutput::DisplayOutput(DisplayOutput&& rhs)
 {
-    std::memcpy(this, &rhs, sizeof(*this));
+    std::memcpy(
+        static_cast<MirDisplayOutput*>(this),
+        static_cast<MirDisplayOutput*>(&rhs),
+        sizeof(MirDisplayOutput));
```

Comment:
Hm. This is invoking undefined behaviour, isn't it? I can see why GCC 8 is complaining about this - we're std::memcpy()ing something that isn't TriviallyCopyable (among other things, it has a non-trivial move constructor *right here* 😁 ), but casting *this into the base class which is TriviallyCopyable doesn't resolve the problem, it just silences the warning, while still letting GCC 8 optimise this code into `rm -rf /`?

This is *somewhat* academic - while there's no guarantee that mcl::DisplayOutput has the same memory-representation as MirDisplayOutput it quite obviously will. Unless we later change mcl::DisplayOutput and forget to update the move constructor.

Reply:
This is fine, we're copying the TriviallyCopyable base member. If there were an implicit_cast<> we would use it. static_cast<> is the next best thing.